### ISSUE TITLE
[Enhancement] Target specific pokemon that spawn nearby

### ISSUE THREAD

**Issue by elmagow** [SwipeX/PokeMate#101]
On poke ++ (jailbroken iphone), the app shows with an icon, pokemons that spawn nearby you. Then you click on it and it run towards the pokemon. It is really usefull and I didn't got softbaned for it as I speak. Catching lot of rare in central park area.

I don't have the coding skills to do it, so maybe someone can make this.

And maybe add an option target/exclude specific pokemon (ids in config)

**Comment by SonnyX** [comment]
I personally also would like an alternative to the current navigator stystem:
Something that runs between pokestops/pokemons, maybe give it a priority system such as: 
if pokestop is nearly in range, it has a higher priority than one further away, multiple pokestops near each other increase each others priority, if it has an active lure-module, increase priority even more. Pokemons add more priority to a stop aswell. If a rare pokemon spawns, give it a higher priority than normal ones (maybe based on the Tier list: https://rankedboost.com/pokemon-go/tier-list/ ).

**Comment by mjmjelde** [comment]
The whole reason behind the current walking system, and what you will lose, is the ability to walk on paths.  This looks less bot like for when they do add checks for bots (if ever).

In that case, I think another walking type should be added to do what you say, not replace the one that is currently there.  Then the user can pick which one they want

**Comment by elmagow** [comment]
Yes, I understand the "human like" walking, that's why I said that I didn't get softbaned for it.

The thing that if you choose to farm an area like central park with a lot of pokemons, pokestops, is that there is no path. So the bot doesn't roam everywhere. 
I don't see how they can differentiate bots from players when for exemple your are in the country, your building, in a park/forest and you open pokemon, there is no road. I doubt they are going to ban everyone that doesn't walk on a road.

**Comment by TVsIan** [comment]
Rather than replace it entirely, perhaps have it following the existing path, then just walking straight to the Pokemon/Pokestop if it's nearby and back before continuing might work. There are a lot of times I'll see the bot walk past Pokestops that are just off the road without interacting.

Alternatively, an option to import a GPX file (or some other GPS route file) instead of letting the bot generate its own path would be nice for parks or other areas where there are a lot of things off of any mapped path.

**Comment by SonnyX** [comment]
I would still like the option to choose between the way it navigates.
Personally I would really like the idea of the bot to become smarter by using the priority system I described above, this should/could also increase the XP per hour massively plus get more of the pokemons that I prioritize (for example overide the standard priority of the magicarp to a really high one because I want a Gyrados).
